python/importer/main.py

```diff
@@ -0,0 +1,86 @@
+    conn = psycopg.connect(**get_conn_params())
+    try:
+        with conn:
+            with conn.cursor() as cur:
+                imported_successfully = import_data_to_db(cur=cur)
+                print(f' imported_successfully {imported_successfully}')
+    finally:
+        conn.close()
+        if imported_successfully == True:
+            print("Import done successfully, running analysis.")
+            run_analysis()
+
+def import_data_to_db(cur):
+    yesterday = datetime.now() - timedelta(1)
+    yesterday = datetime.strftime(yesterday, '%Y-%m-%d')
+    hfp_storage_container_name = os.getenv('HFP_STORAGE_CONTAINER_NAME')
+    hfp_storage_connection_string = os.getenv('HFP_STORAGE_CONNECTION_STRING')
+    service = BlobServiceClient.from_connection_string(conn_str=hfp_storage_connection_string)
+    result = service.find_blobs_by_tags(f"@container='{hfp_storage_container_name}' AND min_oday <= '{yesterday}' AND max_oday >= '{yesterday}'")
+
+    blob_names = []
+    for i, r in enumerate(result):
+        # File format is e.g. 2022-05-11T00-3_ARR.csv.zst
+        # extract event type as we know what the format is:
+        type = r.name.split('T')[1].split('_')[1].split('.')[0]
+        if type in event_types_to_import:
+            print(r.name)
+            blob_names.append(r.name)
+
+    imported_successfully = True
+
+    blob_index = 0
+    try:
+        for blob in blob_names:
+            print(f'Importing blob: {blob}')
+            blob_client = service.get_blob_client(container="hfp-v2-test", blob=blob)
+            storage_stream_downloader = blob_client.download_blob()
+            read_imported_data_to_db(cur=cur, downloader=storage_stream_downloader)
+            blob_index += 1
+            # Limit downloading all the blobs when developing.
+            if os.getenv('IS_DEBUG') == 'True' and blob_index > 5:
+                return imported_successfully
+    except Exception as e:
+        imported_successfully = False
+        print(f'Error in reading blob chunks: {e}')
+
+    return imported_successfully
+
+def read_imported_data_to_db(cur, downloader):
+    compressed_content = downloader.content_as_bytes()
+    reader = zstandard.ZstdDecompressor().stream_reader(compressed_content)
+    bytes = reader.readall()
+    hfp_dict_reader = csv.DictReader(StringIO(bytes.decode('utf-8')))
+    import_io = StringIO()
+
+    invalid_row_count = 0
+    for hfp in hfp_dict_reader:
+        if hfp["vehicleNumber"] != None:
+            row = f'{hfp["tst"]},{hfp["eventType"]},{hfp["receivedAt"]},{hfp["ownerOperatorId"]},{hfp["vehicleNumber"]},{hfp["mode"]},{hfp["routeId"]},{hfp["dir"]},{hfp["oday"]},{hfp["start"]},{hfp["oper"]},{hfp["odo"]},{hfp["drst"]},{hfp["locationQualityMethod"]},{hfp["stop"]},{hfp["longitude"]},{hfp["latitude"]}\n'
+            import_io.write(row)
+        else:
+            invalid_row_count += 1
+    # TODO: log invalid row count into db
+    print(f'Import invalid row count: {invalid_row_count}')
```

Comment:
Here, we could specify the reason for invalid rows to achieve a clearer log entry, e.g. `f'Rows not imported due to missing tst, operatorId, or vehicleNumber value: {invalid_row_count}'`. I'd also consider logging this only conditionally, i.e. `if invalid_row_count > 0`.